day3/src/main.rs

```diff
@@ -32,7 +31,27 @@ fn priority(char: char) -> u32 {
     }
 }
 
-fn puzzle2(input: &str) -> u32 {
```

Comment:
@AaronCQL tried the FP approach, lmk if there are any ways I can improve it
one question I have is if there is an implementation that does not require `clone()`ing the intersection hashset and `common_letter`
I had to add them to satisfy the borrow checker